### ISSUE TITLE
Prompt for upload of alternative media files in submission form

### ISSUE THREAD

**Issue by tdonohue** [DSpace/dspace-angular#4438]
## Describe the bug
Related to #3807 as it came up in discussion of that ticket in a Developer Meeting.

In #3807, we'll have a way to prompt users for alternative media *text / descriptions* in the submission form, and store those values in metadata fields like `dc.description.transcript`, `dc.description.audiodescription` or `dc.description.textalternative`.

However, we've realized that a future improvement would be to also prompt the user to upload alternative files, especially transcript and caption files:

* Transcript should be a separate file to upload. Transcripts can be very long, so they may not be ideal for metadata fields. Additionally, if an Item has multiple audio files, then you'd need multiple transcripts (and a way to link each transcript to the correct audio file)
* Possibly also finding a way to improve our current [caption file upload process](https://wiki.lyrasis.org/display/DSDOC7x/User+Interface+Configuration#UserInterfaceConfiguration-Uploadingvideocaptioningfiles) (which is very manual). Caption files and transcript files should likely be treated similarly, and we should prompt for both in the submission form, and "link" them up to the corresponding video/audio file.

## Expected behavior
The submission form should prompt users for a transcript file when uploading an audio file, and a caption file when uploading a video file.  When the transcript or caption is uploaded, they should be "linked" to the corresponding media file.   This "link" could be a formal link, or it could just be linking them by automatically naming them similar (like the current [caption file upload process](https://wiki.lyrasis.org/display/DSDOC7x/User+Interface+Configuration#UserInterfaceConfiguration-Uploadingvideocaptioningfiles) where a caption file must have a specific name to link it to the video file).

## Related work
Depends on #3807 first being implemented, as this issue would extend that work.
